imap_processing/spice/time.py

```diff
@@ -90,3 +90,75 @@ def _sct2e_wrapper(
         return np.array([spice.sct2e(IMAP_SC_ID, s) for s in sclk_ticks])
     else:
         return spice.sct2e(IMAP_SC_ID, sclk_ticks)
+
+
+@typing.no_type_check
+@ensure_spice
+def str_to_et(
+    time_str: Union[str, list[str]],
+) -> Union[float, np.ndarray]:
+    """
+    Convert string to ephemeris time.
+
+    Decorated wrapper for spiceypy.str2et that vectorizes the function in addition
+    to wrapping with the @ensure_spice automatic kernel furnishing functionality.
+    https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.str2et
+
+    Parameters
+    ----------
+    time_str : str or Collection[str]
+        Input string(s) to be converted to ephemeris time.
+
+    Returns
+    -------
+    ephemeris_time: np.ndarray
+        Ephemeris time, seconds past J2000.
+    """
+    if isinstance(time_str, list):
+        return np.array([spice.str2et(t) for t in time_str])
+    else:
+        return spice.str2et(time_str)
+
+
+@typing.no_type_check
+@ensure_spice
+def et_to_utc(
+    et: Union[float, np.ndarray],
+    format_str: str = "ISOC",
+    precision: int = 3,
+    utclen: int = 24,
+) -> Union[str, Collection[str]]:
```

Comment:
Flip the et type hint with the output type hint.
```suggestion
def et_to_utc(
    et: Union[float, Iterable[float]],
    format_str: str = "ISOC",
    precision: int = 3,
    utclen: int = 24,
) -> Union[str, np.ndarray]]:
```
BTW, I see that spiceypy uses `Iterable` rather than `Collection`. That is probably the correct type hint here though they are subtly different.